watertap/unit_models/cstr_injection.py

```diff
@@ -302,6 +311,56 @@ def cstr_performance_eqn(b, t, r):
         ):
             self.deltaP = Reference(self.control_volume.deltaP[:])
 
+        self.electricity_consumption = Var(
+            self.flowsheet().time,
+            units=pyunits.kW,
+            bounds=(0, None),
+            doc="Electricity consumption of unit",
+        )
+        # The value is taken from Maravelias' data
+        self.energy_electric_flow_vol_inlet = Param(
+            initialize=0.212845,
+            units=pyunits.kWh / pyunits.m**3,
+            mutable=True,
+            doc="Electricity intensity with respect to inlet flow",
+        )
+
+        self.hydraulic_retention_time = Var(
+            self.flowsheet().time,
+            initialize=1460.407,
+            domain=NonNegativeReals,
+            units=pyunits.s,
+            doc="Hydraulic retention time",
+        )
+
+        def CSTR_injection_retention_timee_rule(self, t):
+            return (
+                self.hydraulic_retention_time[t]
+                == self.volume[t] / self.control_volume.properties_in[t].flow_vol
+            )
```

Comment:
Is it necessary to have the HRT defined as a variable and then also calculate it in a constraint? In my CSTR PR, I just define HRT as a mutable parameter (although maybe I should make it a variable).

Reply:
I defined it as a variable and constraint to be able to get calculated or set. I.e., in the BSM2 model, we have the volume and flow, so you can calculate HRT and not give it an actual value. On the other hand, if you want to calculate volume from flow and HRT, the constraint would calculate it. If I had set it as a parameter, I would have to give it a value which, like in the case of BSM2, the HRT is not exact, or I may not be required to know it.

Reply:
I would also opt for defining HRT as a variable with an associated constraint.